### ISSUE TITLE
Question: Use as a filter for files

### ISSUE THREAD

**Issue by nilp0inter** [werman/noise-suppression-for-voice#42]
Hi,

it would be very nice to have the possibility to have a binary that can be used to process noise audio files and produce clean output files.

Personally I'd find it very useful to cleanup audio from lectures on Youtube.

Do you think is doable?  Do you have any plans of implementing such a feature?  If not, could you point me on how to do it?

**Comment by theglobe** [comment]
You can use the plugins in many audio/video editing programs.

For example, I use the VST plugin Davinci Resolve.

**Comment by nilp0inter** [comment]
Unfortunately I can't find a way  to use this from the command line. Do you know of any software to do that?

**Comment by theglobe** [comment]
I only use GUI based software, but maybe ffmpeg can transcode with the ladspa plugin.

**Comment by richardpl** [comment]
FFmpeg also have native filter, arnndn, you need to get rnnoise model files though.

**Comment by jagger2048** [comment]
For the command-line tool. The original [rnnoise repo](https://github.com/xiph/rnnoise) may help you.

**Comment by werman** [comment]
> maybe ffmpeg can transcode with the ladspa plugin.

Yes, I think ffmpeg should be able to do this, see https://ffmpeg.org/ffmpeg-filters.html#ladspa

**Comment by realies** [comment]
You might have to compile FFmpeg with LADSPA support.

**Comment by DarkAlchy** [comment]
> 
> 
> You can use the plugins in many audio/video editing programs.
> 
> For example, I use the VST plugin with Davinci Resolve.

How?  Most times it shows it loaded fine but I can't find it and resolve (Fairlight) is lame in it has to way to rescan your VSTs.  When Resolve Studio 16 finally did show it for me it created a wide stereo and no reduction.

**Comment by THS-on** [comment]
You can use the ladspa plugin directly with mpv:
```
mpv --af=lavfi='[ladspa=librnnoise_ladspa:p=noise_suppressor_mono]' file.mp4
```
This then should work with the youtube-dl integration of mpv.
